security/pfSense-pkg-tinc/files/usr/local/www/status_tinc.php

```diff
@@ -21,78 +21,54 @@
 
 require("guiconfig.inc");
 
-function tinc_status_usr1() {
-	exec("/usr/local/sbin/tincd --config=/usr/local/etc/tinc -kUSR1");
+function tinc_status_usr($usr) {
+	exec("/usr/local/sbin/tincd --config=/usr/local/etc/tinc -k{$usr}");
 	usleep(500000);
-	$clog_path = "/usr/local/sbin/clog";
 	$result = array();
 	$logfile = "/var/log/tinc.log";
+	$clog_path = "/usr/local/sbin/clog";
 	$firmware = host_firmware_version();
-
-	if ($firmware['config_version'] >= 19.7) {
-		exec(system_log_get_cat() . ' ' . $logfile . "| /usr/bin/sed -e 's/.*tinc\[.*\]: //'", $result);
-	} else {
-		exec("{$clog_path} /var/log/tinc.log | /usr/bin/sed -e 's/.*tinc\[.*\]: //'", $result);
+	
+	if ($firmware['kernel']['version'] >= 12.0) {
+		exec(system_log_get_cat() . ' ' . sort_related_log_files($logfile, true, true) . "| /usr/bin/sed -e 's/.*tinc\[.*\]: //'", $result);
+		} else {
+		exec("{$clog_path} {$logfile} | /usr/bin/sed -e 's/.*tinc\[.*\]: //'", $result);
 	}
 	
 	$i = 0;
-	foreach ($result as $line) {
-		if (preg_match("/Connections:/", $line)) {
-			$begin = $i;
+	if ($usr == 'USR1') {
+		foreach ($result as $line) {
```

Comment:
The USR1 and else block are still pretty similar. Looks like you could use the `if ($usr == 'USR1') ... else` test to set the beginning and ending match strings only, and then just have one `foreach`.